sssd_test_framework/markers.py

```diff
@@ -22,10 +24,17 @@ def pytest_configure(config: pytest.Config):
     )
 
 
-def builtwith(item: pytest.Function, requirements: dict[str, str], **kwargs: BaseRole):
+def builtwith(item: pytest.Function, requirements: dict[str, str], topology: KnownTopology, **kwargs: BaseRole):
```

Comment:
this should take type `Topology | None`, not `KnownTopology`

Reply:
Also add `mh: MultihostFixture` to parameters and ditch `mh = item.funcargs["mh"]`

Reply:
I kept this part as it was before the PR because I moved all the checks to `pytest_runtest_setup()`, as you suggested in another comment